src/Engine/GameObject/GameObject.cpp

```diff
@@ -27,6 +30,10 @@ void GameObject::notify_destroyed() {
     }
 }
 
+void GameObject::take_damage(int damage) {
+    health_ -= damage;
```

Comment:
I think notify_destroyed if health < 0 should be the default behaviour. How targetting system is doing things is not the normal use case and should be the one overriding this function, not basic enemy or main character.